### PR TITLE
Rename metadata interface ix fields from `token_program_id` to `program_id`

### PR DESCRIPTION
**Pull request by jacksondoherty**
See #3073.

Rename fields to better reflect interface structure and match [spl_token_metadata_interface](https://docs.rs/spl-token-metadata-interface/latest/spl_token_metadata_interface/index.html) and [spl_token_group_interface](https://docs.rs/solarti-token-group-interface/latest/spl_token_group_interface/index.html) instructions.

### CODE REVIEW

**Review by acheroncrypto — COMMENTED** [review]
Thanks! We can get this in before the next breaking release. Holding off for now to give us the option of a patch release in-between.

**Review by acheroncrypto — COMMENTED** [review]
We don't need a patch release since the upstream SPL issue regarding Solana v2 (https://github.com/coral-xyz/anchor/issues/3044) has been fixed with the new release of SPL crates.

Before we merge, could you note this breaking change in the [CHANGELOG](https://github.com/coral-xyz/anchor/blob/f677742a978ffdf7bc321746b4119394f6654b7c/CHANGELOG.md#breaking)?

**Comment by jacksondoherty** [comment]
Sorry, lost track of this – breaking note added to CHANGELOG + rebased with latest changes from master 👍

File: CHANGELOG.md
Comment (acheroncrypto) [inline CHANGELOG.md:60]: Missing "#":

```suggestion
- spl: Rename metadata interface instruction fields from `token_program_id` to `program_id` ([#3076](https://github.com/coral-xyz/anchor/pull/3076)).
```

**Review by acheroncrypto — APPROVED** [review]
Thanks!

File: CHANGELOG.md
Comment (acheroncrypto) [reply CHANGELOG.md:60]: Nevermind — I was going to commit this before merging, but couldn't due to this branch belonging to an organization's fork

**Pull request merged**
